modules/internal/String.chpl

```diff
@@ -151,6 +152,10 @@ module String {
     var shortData : chpl__inPlaceBuffer;
   }
 
+  record codePointIndex {
+    var cpidx     : int;
+  }
```

Comment:
Let's add _cast functions so that myCodePointIndex:int and myInt:codePointIndex will work.